nodeadm/internal/configprovider/userdata.go

```diff
@@ -17,22 +18,41 @@ import (
 
 const (
 	contentTypeHeader          = "Content-Type"
+	contentEncodingHeader      = "Content-Encoding"
 	mimeBoundaryParam          = "boundary"
 	multipartContentTypePrefix = "multipart/"
 	nodeConfigMediaType        = "application/" + api.GroupName
 )
 
-type userDataConfigProvider struct{}
+type userDataProvider interface {
+	GetUserData() ([]byte, error)
+}
+
+type imdsUserDataProvider struct{}
+
+func (p *imdsUserDataProvider) GetUserData() ([]byte, error) {
+	return imds.GetUserData()
+}
+
+type userDataConfigProvider struct {
+	userDataProvider userDataProvider
+}
 
 func NewUserDataConfigProvider() ConfigProvider {
-	return &userDataConfigProvider{}
+	return &userDataConfigProvider{
+		userDataProvider: &imdsUserDataProvider{},
+	}
 }
 
 func (ics *userDataConfigProvider) Provide() (*internalapi.NodeConfig, error) {
```

Comment:
Why is `ics` the name for `userDataConfigProvider`?

Reply:
i think it's an artifact from when it was something like `imdsConfigProvider`? 🤣   
maybe we could just generally name the config provider references `cp`

Reply:
tbh i like just calling it `this` but that's a Java-ism that gophers detest

Reply:
Haha. Let's change it to something that makes sense :P